### ISSUE TITLE
Training strategy of deeplabv3 on the dataset

### ISSUE THREAD

**Issue by gavinwxy** [skokec/segdec-net-jim2019#23]
Hi,

Thanks for sharing your great work! I encountered problems when training with deeplabv3 on your dataset. 

My training setting is to use original labels without dilation and deeplabv3 resnet101. The model is trained on a training set consist of  40 images with defects and 40 clean images and is validated on a validation set with only 12 defect images.

The problem is deeplabV3 seems to overfit the data heavily (reaches 0.80+ IoU) and cannot predict on validation set (nearly 0 IoU). In your paper, the result of deeplabv3 as a baseline seems pretty good. Could you share your training strategy or give me some advice on training. Thanks in advance!

**Comment by skokec** [comment]
Hi, 

For DeepLab v3+ the overfitting issue is expected due to large number of parameters. In your case it might be even more pronounced due small number of clean images, making the optimization problem severely over-determined.

I am not sure if any training strategy might mitigate this problem. You can give L1 or L2 regularization a try, but I think using DeepLab for problems with small number of training samples is an overkill and smaller networks should work better. If you still want to use DeepLab architecture then try the one with the smallest backbone, e.g., MobileNet. 

Best,
Domen

**Comment by gavinwxy** [comment]
Thanks for your reply!
